src/main/scala/beam/router/BeamSkimmer.scala

```diff
@@ -1,19 +1,41 @@
 package beam.router
+
 import beam.agentsim.agents.choice.mode.DrivingCost
 import beam.agentsim.agents.vehicles.BeamVehicleType
+import beam.agentsim.infrastructure.TAZTreeMap.TAZ
 import beam.router.BeamRouter.Location
+import beam.router.BeamSkimmer.Skim
 import beam.router.Modes.BeamMode
-import beam.router.model.{BeamLeg, BeamPath}
+import beam.router.Modes.BeamMode.{
+  BIKE,
+  CAR,
+  CAV,
+  DRIVE_TRANSIT,
+  RIDE_HAIL,
+  RIDE_HAIL_POOLED,
+  RIDE_HAIL_TRANSIT,
+  TRANSIT,
+  WALK,
+  WALK_TRANSIT
+}
+import beam.router.model.{BeamLeg, BeamPath, EmbodiedBeamTrip}
 import beam.sim.BeamServices
 import beam.sim.common.GeoUtils
+import beam.utils.FileUtils
+import com.google.inject.{Inject, Provider}
+import org.matsim.api.core.v01.Id
+import org.matsim.core.controler.events.IterationEndsEvent
+import org.matsim.core.controler.listener.IterationEndsListener
 
-case class TimeDistanceAndCost(timeAndCost: TimeAndCost, distance: Option[Int]) {
-  override def toString =
-    s"[time:${timeAndCost.time.getOrElse("NA")}][cost:${timeAndCost.cost.getOrElse("NA")}][distance:${distance.getOrElse("NA")}]"
-}
+import scala.collection.concurrent.TrieMap
 
 //TODO to be validated against google api
-class BeamSkimmer(services: Option[BeamServices] = None) {
+class BeamSkimmer @Inject()() extends IterationEndsListener {
+  // The OD/Mode/Time Matrix
+  var previousSkims: TrieMap[(Int, BeamMode, Id[TAZ], Id[TAZ]), Skim] = TrieMap()
```

Comment:
should a case class be used instead of 4-tuple key, e.g. as a refactor after merge? / same for var skims

Reply:
That would make things more clear. Let's do that when we expand what is stored in the skimmer.